questionpy_server/worker/impl/_base.py

```diff
@@ -101,7 +114,9 @@ def send(self, message: MessageToWorker) -> None:
             raise WorkerNotRunningError
         self._connection.send_message(message)
 
-    async def send_and_wait_for_response(self, message: MessageToWorker, expected_response_message: type[_M]) -> _M:
+    async def send_and_wait_for_response(
+        self, message: MessageToWorker, expected_response_message: type[_M], timeout: float | None = None
+    ) -> _M:
```

Comment:
Dadurch leckt das Timeout-Feature ein wenig "nach oben" und in den ThreadWorker, der damit nichts anfangen kann. Wäre es nicht besser, im Mixin `_get_observation_tasks` und `send_and_wait_for_response` zu overriden? Für `_init_worker_timeout` und `_load_qpy_package_timeout` könnte in letzterem entweder anhand des Typs von `message` der Timeout ausgewählt oder eigene Methoden erstellt werden.

Reply:
Finde ich eigentlich nicht so schlimm, besonders weil der ThreadWorker nicht für den Produktivbetrieb gedacht ist.